src/stdlib/src/lib.rs

```diff
@@ -396,8 +400,19 @@ pub extern "C" fn srandom(seed: c_uint) {
 }
 
 #[no_mangle]
-pub extern "C" fn strtod(s: *const c_char, endptr: *mut *mut c_char) -> c_double {
-    unimplemented!();
+pub unsafe extern "C" fn strtod(s: *const c_char, endptr: *mut *mut c_char) -> c_double {
+    //TODO: endptr
+
+    use core::str::FromStr;
+
+    let s_str = str::from_utf8_unchecked(platform::c_str(s));
+    match f64::from_str(s_str) {
```

Comment:
Nice! Should `atoi` and `atol` be updated to use `i32::from_str` and `i64::from_str`. I just realized I made those functions a lot harder than they needed to be.

Reply:
Maybe so, but then we cannot fill in the endptr correctly in strtol and strtoul

Reply:
ah, true... also do the generic `from_str` implementations handle strings like " \t<number>\t"?

Reply:
I don't know. I think the ideal thing is to find the number string, along with its length, then pass that to the from_str handler.